src/components/MAPI/apps/utils.ts

```diff
@@ -914,7 +944,12 @@ export async function getUpgradableApps(
  * @param apps
  */
 export function getUpgradableAppsKey(apps: applicationv1alpha1.IApp[]) {
-  return apps.reduce((key, app) => key + app.spec.name, '');
+  if (apps.length === 0) return null;
+
```

Comment:
Small change here - if there are no installed apps, there is no point in fetching upgradable apps. We return `null` here for the key, so the `useSWR` hook will not execute a request when there are no installed apps.